pkg/synapse/synapse.go

```diff
@@ -37,44 +46,102 @@ func New(
 	logger lumber.Logger,
 	secretsManager core.SecretsManager,
 ) core.SynapseManager {
+
 	return &synapse{
-		runner:         runner,
-		logger:         logger,
-		secretsManager: secretsManager,
+		runner:                   runner,
+		logger:                   logger,
+		secretsManager:           secretsManager,
+		MsgErrChan:               make(chan struct{}),
+		InvalidConnectionRequest: make(chan struct{}),
+		MsgChan:                  make(chan []byte, 1024),
+		ConnectionAborted:        make(chan struct{}, 10),
+		LogoutRequired:           true,
 	}
 }
 
 func (s *synapse) InitiateConnection(
 	ctx context.Context,
 	wg *sync.WaitGroup,
-) {
+	connectionFailed chan struct{}) {
 	defer wg.Done()
+	go s.openAndMaintainConnection(ctx, connectionFailed)
+	<-ctx.Done()
```

Comment:
not getting what you're trying to do here. Passing `ctx` to a go-routine and then immediately sending a Done signal. Won't it cause issues in go-routine where ctx is being used?

Reply:
`<-ctx.Done()` will receive channel message 
both  openAndMaintainConnection and InitiateConnection will listen  for done channel